src/Nethermind/Nethermind.TxPool/Collections/PersistentBlobTxDistinctSortedPool.cs

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Nethermind.Core;
+using Nethermind.Core.Caching;
+using Nethermind.Core.Crypto;
+using Nethermind.Logging;
+
+namespace Nethermind.TxPool.Collections;
+
+public class PersistentBlobTxDistinctSortedPool : BlobTxDistinctSortedPool
+{
+    private readonly ITxStorage _blobTxStorage;
+    private readonly LruCache<ValueKeccak, Transaction> _blobTxCache;
+    private readonly ILogger _logger;
+
+    public PersistentBlobTxDistinctSortedPool(ITxStorage blobTxStorage, ITxPoolConfig txPoolConfig, IComparer<Transaction> comparer, ILogManager logManager)
+        : base(txPoolConfig.PersistentBlobStorageSize, comparer, logManager)
+    {
+        _blobTxStorage = blobTxStorage ?? throw new ArgumentNullException(nameof(blobTxStorage));
+        _blobTxCache = new(txPoolConfig.BlobCacheSize, txPoolConfig.BlobCacheSize, "blob txs cache");
+        _logger = logManager?.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
+
+        RecreateLightTxCollectionAndCache(blobTxStorage);
```

Comment:
if we save SenderTx+Tx we could think about saving transactions in form Tx + SenderTx + Timestamp and during reloading we might add heuristic for deleting super old blobs. Something to think about - out of the scope of this PR

Reply:
I already added timestamp to db 😄 basing on earlier discussion.
Mechanism for removing old blobs can be added in separate PR. I think it will be the best to do it when updating GasBottleneck - just check timestamp and remove if tx is too old (we are doing it by yield returning null). What we still need to do is to figure out what will be the best lifetime for blob txs and to test it well